### ISSUE TITLE
Locales not found when running on iOS

### ISSUE THREAD

**Issue by bhyip23** [xleon/I18N-Portable#34]
I am able to deploy on Android but on iOS, I get the following error:

`I18NPortable.I18NException: No locales have been found. Make sure you've got a folder called 'Locales' containing embedded resource files (with extensions .txt) in the host assembly`

`The app has been terminated.`

I have a Locales folder in my Core project that contains en.txt and fr.txt.

The exception occurs in App.xaml.cs where I18N gets initialized.

```
public App()
{
    InitializeComponent();

    I18N.Current
        .SetLogger(text => Debug.WriteLine(text))
        .SetFallbackLocale("en")
        .Init(GetType().GetTypeInfo().Assembly);

    MainPage = new NavigationPage(new LoginPage());
}
```

**Comment by xleon** [comment]
You need to set Build Action to Embedded Resource on the properties of every .txt file

**Comment by bhyip23** [comment]
Yes, the build action is set to Embedded resource for both my en.txt and fr.txt

![image](https://user-images.githubusercontent.com/3255191/36619318-ac4a7bf8-18ab-11e8-880b-a40a4986e5a4.png)

**Comment by xleon** [comment]
I have the very same setup and it works fine. 
Are you sure that `GetType().GetTypeInfo().Assembly` is the same assembly where .txt files are?

Can you attach a test project that I can try myself?

**Comment by eggf** [comment]
I got similar problem with iOS devices. I18N-Portable does not choose the correct translation file.
For me it seems that the system has mixed up localization codes as you can see in the image.

![unbenannt](https://user-images.githubusercontent.com/5421541/36721165-0ea3748c-1ba2-11e8-93c4-e86061e5afed.JPG)

**Comment by xleon** [comment]
Interesting. That seams like a bug with french locales. 
Can you guys confirm it just happens with french? @eggf @bhyip23

**Comment by eggf** [comment]
No, it is not just with French. Also with Russian, Spanish or Italian language.

**Comment by xleon** [comment]
Ok, I´ll make some tests to see if I can find a fix or at least to isolate the bug (not sure if this is an iOS bug or a Xamarin bug)

**Comment by xleon** [comment]
Can you guys tell what version/s of iOS and Xamarin.iOS are giving you trouble?

**Comment by eggf** [comment]
Visual Studio 2017 Version 15.4.5
Xamarin 4.7.10.38 (34c1b6c)
Xamarin.iOS and Xamarin.Mac SDK 11.3.0.47 (51128b8)
Visual Studio for Mac Version 7.2.2 (build 11)
XCODE Version 9.1 (9B55)

**Comment by xleon** [comment]
Are you testing on simulator or physical device?

**Comment by eggf** [comment]
On both with same errors.
iOS 11.1

**Comment by xleon** [comment]
I think we are hitting this exact big on iOS 11: https://bugzilla.xamarin.com/show_bug.cgi?id=59596

**Comment by bhyip23** [comment]
Just an update for my original problem... I tried to reproduce it with a new project but everything works, so I just moved all my files to the new project.

**Comment by xleon** [comment]
Please @bhyip23 can you attach your old and new info.plist files in the iOS project?

**Comment by clovs** [comment]
Hello
I tried to make it works on Visual studio for Mac.
Seems that ever in Android or iOS ,
I18NPortable throw an exception: No locales have been found. Make sure you've got a folder called 'Locales' containing embedded resource files (with extensions .txt) in the host assembly

Hope this will be fix soon :)

**Comment by bhyip23** [comment]
Sorry for the late reply, been really busy with other stuff.

The only difference in the info.plist is that the old one had an extra entry:
<key>CFBundleName</key>
<string>My_Application</string>
